pkg/api/v1/schema.go

```diff
@@ -35,8 +35,20 @@ type DomainSpec struct {
 }
 
 type Memory struct {
```

Comment:
For this initial patch, It might make this easier to get merged if you limited the scope of initial change to just the "Allocated" field.  Then follow up with the Required and Maximum fields afterwards in another PR. 

I have a feeling those two new values are going to spark a large discussion.

Reply:
It would be best if you follow/use only fields supported in libvirt domain XML https://libvirt.org/formatdomain.html#elementsMemoryAllocation

Reply:
I can't, because it does not contain the necessary fields. It also does not say anything about libvirt XML in v1.Domain so it is kind of hard to guess limitations like this.

Reply:
so which exactly are you missing? just the "required" mentioned below?

Reply:
We don't have to maintain a 1:1 relationship between the KubeVirt domain spec and libvirt xml. 

Like @fabiand mentioned, if possible we should consider using k8s terminology for this feature as it makes sense.  The "allocated" and "maximum" fields appear to map to the k8s "request" and "limit" concepts. The new "required" field is it's own concept entirely though.

Reply:
> We don't have to maintain a 1:1 relationship between the KubeVirt domain spec and libvirt xml.

we definitely should. Not the POD definition, sure, we want to use it as much as possible where we an, but the VM spec side should be 1:1 except for the host specific bits. That's not this case

Reply:
Ok, so I moved the specific bits from Domain to a separate structure within VMSpec in hope it would make this cleaner.

Reply:
> we definitely should. Not the POD definition, sure, we want to use it as much as possible where we can, but the VM spec side should be 1:1 except for the host specific bits. That's not this case

This appears inconsistent with the direction we're headed right now. You may want to bring this up on the mailing list for a more detailed discussion.

Reply:
should be addressed by #391